cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1572,7 +1652,80 @@ void writer::impl::write(table_view const& table, std::vector<partition_info> co
     }
   }
 
-  // Build chunk dictionaries and count pages
+  // until now we've been using a uniform fragment size. if there are any columns with
+  // a fragment size != max_page_fragment_size_ then resize the fragments array and
+  // update chunks with new fragment info
+
+  // first figure out the total number of fragments
+  // calculate start offset for each column
+  std::vector<size_type> frag_offsets;
+  size_type total_frags = 0;
+
+  if (frags_per_column.size() > 0) {
+    std::exclusive_scan(frags_per_column.data(),
+                        frags_per_column.data() + num_columns + 1,
+                        std::back_inserter(frag_offsets),
+                        0);
+    total_frags = frag_offsets[num_columns];
+  }
+
+  bool const is_resize_fragments = total_frags != num_columns * num_fragments;
```

Comment:
I would be okay with the assumption that this is always true. Seems like a low overhead to help avoid some of the complexity below.

Reply:
The fragment code is one of the few things in the parquet writer that is insanely parallel.  I surrender :D